### ISSUE TITLE
New Rule: Transactions in a function should be passed through all queries. 

### ISSUE THREAD

**Issue by waleedwaseem** [eslint/eslint#15573]
### Rule details

It should make sure that once the transaction is declared on top, it should pass all the queries in that function else it can choke the node server and has done multiple times. 

### What type of rule is this?

Warns about a potential problem

### Example code

```js
create: async (req, res) => {
  const transaction = await sequelize.transaction();
  try {
    let user = await User.findOne(req.body.id); // no transaction passed should throw the error.
    if (!user)
      await User.create(user, { transaction });
    await transaction.commit() 
    res.status(200);
  } catch {
     await transaction.rollback();
     res.status(400);
  }
```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by golopot** [comment]
This a good rule but it should be implemented in a plugin like `eslint-plugin-sequalize`.

**Comment by nzakas** [comment]
Yes, ESLint core rules can’t use any libraries or frameworks. Any rules specific to libraries or frameworks should be in a plugin.

**Comment by waleedwaseem** [comment]
@nzakas , Can you kindly guide me on how i can initiate a request for that?
